### PR TITLE
[EGGO-44] Use Cloudera Director for cluster provisioning.

### PR DESCRIPTION
**Pull request by tomwhite**
This change makes it possible to start and stop a cluster using Cloudera
Director. However, it is not tied in to Eggo operations yet.

### CODE REVIEW

File: conf/aws.conf [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+    # See: http://docs.aws.amazon.com/AmazonVPC/latest/UserGuide/VPC_SecurityGroups.html
+    #
+
+    securityGroupsIds: %(securityGroupsIds)s
+
+    #
+    # A prefix that Cloudera Director should use when naming the instances (this is not part of the hostname)
+    #
+
+    instanceNamePrefix: cloudera-director
```
Comment (laserson) [inline conf/aws.conf:75]: nit: change to `eggo-director`?

File: conf/aws.conf [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+            owner: %(keyName)s
+        }
+
+        bootstrapScript: """#!/bin/sh
+
+# This is an embedded bootstrap script that runs as root and can be used to customize
+# the instances immediately after boot and before any other Cloudera Director action
+
+# If the exit code is not zero Cloudera Director will automatically retry
+
```
Comment (laserson) [inline conf/aws.conf:152]: Should we put some of our eggo-specific init here?  Or continue to leave it to fabric?  Probably the latter...

File: eggo/director.py [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+def list():
+    conn = create_ec2_connection()
+    print 'Launcher', get_launcher_instance(conn).ip_address
+    print 'Manager', get_manager_instance(conn).ip_address
+    print 'Gateway', get_gateway_instance(conn).ip_address
+    print 'Master', get_master_instance(conn).ip_address
+
+def login():
+    conn = create_ec2_connection()
+    hosts = get_gateway_instance(conn).ip_address
```
Comment (laserson) [inline eggo/director.py:62]: Do we generally want to login to the gateway or the master?  I thought the gateway is generally a puny node that is used purely to coordinate the cluster provisioning.

Comment (laserson) [inline eggo/director.py:62]: oh...ugh, launcher, manager, gateway, master instances...

**Comment by laserson** [comment]
This is awesome, @tomwhite! Should we merge it in so easier to iterate?

File: conf/aws.conf (hunk 1 above)
Comment (tomwhite) [inline conf/aws.conf:75]: I was trying to leave this eggo-agnostic, so we can move it out or use it separately.

File: eggo/director.py (hunk 3 above)
Comment (tomwhite) [inline eggo/director.py:62]: The gateway is the machine you run jobs from. The master runs the master daemons, and should generally not be logged in to by users (a good security practice). The manager is the machine running Cloudera Manager, which may be the same as the master. The launcher is an instance used by Cloudera Director to launch the cluster from.

**Comment by tomwhite** [comment]
I merged and opened https://github.com/bigdatagenomics/eggo/issues/48 as a follow on.

**Pull request merged**